src/shims/tls.rs

```diff
@@ -38,6 +38,9 @@ pub struct TlsData<'tcx> {
 
     /// Whether we are in the "destruct" phase, during which some operations are UB.
     dtors_running: HashSet<ThreadId>,
+
+    /// The last TlsKey used to retrieve a TLS destructor.
+    last_dtor_key: BTreeMap<ThreadId, TlsKey>,
```

Comment:
This needs a bit more explanation. I think I can fathom what this is for based on having recently read this code, but the comment is not explaining what that is for and why it is relevant.

I'd probably also rename it or so... something like
```rust
/// The state needed to track which TLS dtor we are currently running for each thread.
tls_dtor_state: BTreeMap<ThreadId, TlsKey>,
```

Reply:
Never mind, see https://github.com/rust-lang/miri/pull/1284#discussion_r415049589.